### PR TITLE
Improvements to the HTTP based span decorators

### PR DESCRIPTION
**Pull request by objectiser**
This PR includes the following:
[x] Updated opentracing-java version to 0.20.10
[x] Use constants for DB tags available in 0.20.10
[x] Changed default operation to be component name
[x] Added mechanism to enable custom decorator for particular component to override standard one
[x] Added span decorators for direct/vm/seda components

### CODE REVIEW

**Comment by objectiser** [comment]
@davsclaus Hi Claus, when you get a chance if you could review please?

File: components/camel-opentracing/src/main/java/org/apache/camel/opentracing/decorators/AbstractSpanDecorator.java [hunk 1]
```diff
@@ -33,12 +33,40 @@
...
+                : endpoint.getEndpointUri().substring(start, end);
     }
 
     @Override
     public void pre(Span span, Exchange exchange, Endpoint endpoint) {
         span.setTag(Tags.COMPONENT.getKey(), CAMEL_COMPONENT + URI.create(endpoint.getEndpointUri()).getScheme());
+
+        // Including the endpoint URI provides access to any options that may have been provided, for
+        // subsequent analysis
+        span.setTag("camel.uri", endpoint.getEndpointUri());
```
Comment (davsclaus) [inline components/camel-opentracing/src/main/java/org/apache/camel/opentracing/decorators/AbstractSpanDecorator.java]: I wonder if we should mask sensitive information such as passwords, which Camel do with

org.apache.camel.util.URISupport#sanitizeUri

which it does by default in its own toString

Comment (objectiser) [inline components/camel-opentracing/src/main/java/org/apache/camel/opentracing/decorators/AbstractSpanDecorator.java]: Sounds like a good idea - I'll update shortly. Thanks.

**Comment by objectiser** [comment]
@davsclaus Thanks! Update applied.

**Comment by davsclaus** [comment]
Thanks for the PR it has been merged. Do you mind closing this?

**Pull request closed without merging**